web/src/app/wizard/WizardScheduleForm.tsx

```diff
@@ -79,27 +95,41 @@ export default function WizardScheduleForm({ value, onChange, secondary }) {
     }
   }
 
-  const getKey = () => {
-    return secondary ? 'secondarySchedule' : 'primarySchedule'
-  }
-
-  const handleRotationTypeChange = (e) => {
+  const handleRotationTypeChange = (
+    e: React.ChangeEvent<HTMLInputElement>,
+  ): void => {
     const newVal = _.cloneDeep(value)
-    newVal[getKey()].rotation.type = e.target.value
+    if (secondary) {
+      newVal.secondarySchedule.rotation.type = e.target.value as RotationType
+    } else {
+      newVal.primarySchedule.rotation.type = e.target.value as RotationType
+    }
     onChange(newVal)
   }
 
-  const handleFollowTheSunToggle = (e) => {
+  const handleFollowTheSunToggle = (
+    e: React.ChangeEvent<HTMLInputElement>,
+  ): void => {
     const newVal = _.cloneDeep(value)
-    newVal[getKey()].followTheSunRotation.enable = e.target.value
+    if (secondary) {
```

Comment:
is secondary and key now redundant?

Reply:
Added a helper function and cleaned up the functions to help reduce the redundancy a bit